modules/component/src/core/cd-aware/cd-aware_creator.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  combineLatest,
+  EMPTY,
+  NEVER,
+  NextObserver,
+  Observable,
+  PartialObserver,
+  Subject,
+  Subscribable,
+  Subscription,
+} from 'rxjs';
+import {
+  catchError,
+  distinctUntilChanged,
+  map,
+  startWith,
+  switchMap,
+  tap,
+} from 'rxjs/operators';
+import {
+  CdStrategy,
+  DEFAULT_STRATEGY_NAME,
+  StrategySelection,
+} from './strategy';
+
+export interface CdAware<U> extends Subscribable<U> {
+  nextValue: (value: any) => void;
+  nextConfig: (config: string) => void;
+}
+
+/**
+ * class CdAware
+ *
+ * @description
+ * This abstract class holds all the shared logic for the push pipe and the let directive
+ * responsible for change detection
+ * If you extend this class you need to implement how the update of the rendered value happens.
+ * Also custom behaviour is something you need to implement in the extending class
+ */
+export function createCdAware<U>(cfg: {
+  strategies: StrategySelection<U>;
+  resetContextObserver: NextObserver<any>;
+  updateViewContextObserver: PartialObserver<U> & NextObserver<U>;
+}): CdAware<U | undefined | null> {
+  const configSubject = new Subject<string>();
+  const config$: Observable<CdStrategy<U>> = configSubject.pipe(
+    distinctUntilChanged(),
```

Comment:
May want to move this after `startWith` to ensure that it doesn't fire the same thing twice if the first value from `configSubject` happens to be the same as `startWith`.

Also consider using a `BehaviorSubject` instead of `startWith`? Depends on your goal. It seems like you might end up with a situation where some code is changed or adjusted to subscribe in a more lazy way, and `renderSideEffect$` could be subscribed to *after* the initial value is emitted from here, and it may be confusing to debug.

Reply:
> May want to move this after startWith to ensure that it doesn't fire the same thing twice if the first value from configSubject happens to be the same as startWith.

Done

Reply:
>Also consider using a BehaviorSubject instead of startWith? Depends on your goal. It seems like you might end up with a situation where some code is changed or adjusted to subscribe in a more lazy way, and renderSideEffect$ could be subscribed to after the initial value is emitted from here, and it may be confusing to debug.

Done